src/options.js

```diff
@@ -239,7 +239,47 @@ exports.bool = {
      *             warnings/errors in when upgrading between minor versions of
      *             JSHint.
      */
-    enforceall : false
+    enforceall : false,
+
+    /**
+     * This option warns when a comma is not placed after the last element in an
+     * array or object literal. Due to bugs in old versions of IE, trailing
+     * commas used to be discouraged, but since ES5 their semantics were
```

Comment:
As you mentioned [here](https://github.com/jshint/jshint/pull/2100#discussion_r89679455), the syntax was actually specified before ES5. If we went the "practical" route I suggest [here](https://github.com/jshint/jshint/pull/3090/files#r96153612), we could reword this to, "but since IE implementations of ES5, their semantics became uniform across most JS environments."